apps/www/components/Audio/hooks/useMediaSession.tsx

```diff
@@ -0,0 +1,124 @@
+import { useCallback, useEffect, useRef } from 'react'
+import { CDN_FRONTEND_BASE_URL } from '../../../lib/constants'
+import { getImageCropURL } from '../AudioPlayer/ui/AudioCover'
+import { AudioQueueItem } from '../graphql/AudioQueueHooks'
+
+type PlayerState = {
+  duration: number
+  playbackRate: number
+  currentTime: number
+}
+
+type MediaSessionCallbacks = {
+  onPlay: () => void
+  onPause: () => void
+  onSeekForward: (seekOffset: number) => void
+  onSeekBackward: (seekOffset: number) => void
+  onSkipToNext: () => void
+  onStop: () => void
+  onRetrievePlayerState: () => PlayerState
+}
+
+type MediaSessionOptions = {
+  isPlaying: boolean
+  callbacks: MediaSessionCallbacks
+}
+
+export function useMediaSession(
+  playerItem: AudioQueueItem | null,
+  { isPlaying, callbacks }: MediaSessionOptions,
+) {
+  const callbackRefs = useRef(callbacks)
+
+  useEffect(() => {
+    callbackRefs.current = callbacks
+  }, [callbacks])
+
+  const updatePlayerState = useCallback(() => {
+    if ('mediaSession' in navigator) {
+      const playerState = callbackRefs.current.onRetrievePlayerState()
+      navigator.mediaSession.setPositionState({
+        duration: playerState.duration,
+        playbackRate: playerState.playbackRate,
+        position: playerState.currentTime,
+      })
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!('mediaSession' in navigator)) {
+      return
+    }
+    const mediaSession = navigator.mediaSession
+
+    mediaSession.metadata = new MediaMetadata({
+      title: playerItem.document?.meta?.title || 'Ein Beitrag der Republik',
+      artist: 'Republik Magazin',
+      album: 'Republik «Vorgelesen»',
+      artwork: [
+        getMediaImage(playerItem, 96),
+        getMediaImage(playerItem, 128),
+        getMediaImage(playerItem, 192),
+        getMediaImage(playerItem, 256),
+        getMediaImage(playerItem, 384),
+        getMediaImage(playerItem, 512),
+      ],
+    })
+
+    if (playerItem) {
+      mediaSession.playbackState = isPlaying ? 'playing' : 'paused'
+
+      if (callbackRefs?.current) {
```

Comment:
`callbackRefs` can't be undefined (just `current`)

```suggestion
      if (callbackRefs.current) {
```